src/main/java/whobot/task/Event.java

```diff
@@ -0,0 +1,122 @@
+package whobot.task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import whobot.main.WhoBotException;
+
+/***
+ * Class to Handle Event type Tasks
+ */
+public class Event extends Task {
+
+    /** Timing for the Event */
+    private final LocalDateTime timing;
+
+    /***
+     * Constructor for the Event Class
+     *
+     * @param task the string description of the Event Task
+     * @throws WhoBotException If the format of timing or command is not correct
+     */
+    public Event(String task) throws WhoBotException {
+        super(task.split(" /at ")[0]);
+        try {
+            this.timing = LocalDateTime.parse(task.split(" /at ")[1], DateTimeFormatter.ofPattern("d/M/yyyy HH:mm"));
```

Comment:
Maybe you can use some local (temporary) variables here, since the expression does a lot of computation and is quite complicated.